sshd-core/src/main/java/org/apache/sshd/common/io/nio2/Nio2Session.java

```diff
@@ -70,6 +71,8 @@
     private final AtomicLong lastReadCycleStart = new AtomicLong();
     private final AtomicLong writeCyclesCounter = new AtomicLong();
     private final AtomicLong lastWriteCycleStart = new AtomicLong();
+    private final AtomicBoolean suspendRead = new AtomicBoolean(false);
+    private final AtomicReference<Runnable> readRunnable = new AtomicReference<>(null);
```

Comment:
In the current state, it's not needed imho, because this is only used for "dumb" tcp connections, i.e. there's no graceful shutdown of the sessions, the sockets are simply closed for forwarding sessions.  I've just added some javadoc to make it clear.